backend/src/common/mocks/repository.mock.ts

```diff
@@ -34,8 +34,20 @@ export class MockRepository<T extends Identifiable> {
   findOne = jest.fn((id: number): undefined | T => {
     return this.#entities[id];
   });
+  findOneBy = jest.fn((query: { id: number }): undefined | T => {
+    return this.#entities[query.id];
+  });
+  findOneByOrFail = jest.fn((query: { id: number }): undefined | T => {
+    return this.#entities[query.id];
+  });
   find = jest.fn((): T[] => this.#entities);
+  findAndCount = jest.fn((): [T[], number] => [
+    [...this.#entities],
+    this.#entities.length,
+  ]);
+  findBy = jest.fn((): T[] => this.#entities);
   findAll = jest.fn((): T[] => this.#entities);
+  findAllBy = jest.fn((): T[] => this.#entities);
```

Comment:
```suggestion
  find = jest.fn((): T[] => this.#entities.filter((x) => x !== undefined));
  findAndCount = jest.fn((): [T[], number] => {
    const returnEntities = this.find();
    return [returnEntities, returnEntities.length];
  });
  findBy = this.find;
  findAll = this.find;
  findAllBy = this.find;
```
Otherwise, there is an `undefined` element at index 0.

Reply:
I would also remove from this file `findAllPaginated`.